runtime/doc/map.txt

```diff
@@ -1430,6 +1430,145 @@ Possible values are (second column is the short name used in listing):
     -addr=other		  ?	other kind of range
 
 
+Incremental preview ~
+                                                  *:command-preview* {nvim-api}
+It's possible to allow commands to be incrementally previewed when
+'inccommand' is enabled by using the preview command flag. The preview
+flag may only be used through the Lua API (see |nvim_create_user_command()|).
+
+The preview callback must be a Lua function. When command preview is
+triggered, the preview callback will be called with 3 arguments, an `opts` table
+which is the same as the table passed to |nvim_create_user_command()| callbacks,
+the preview namespace number that's used to set highlights for the preview,
+and the preview buffer number that's used for the preview window when
+`inccommand=split`. Note that if `inccommand=nosplit`, the preview buffer number
+will be `nil`.
+
+In order for command preview to be shown, these steps need to be followed:
+
+1. Modify the current buffer as required for the preview (see
+   |nvim_buf_set_text()| and |nvim_buf_set_lines()|).
+2. If preview buffer is provided, add necessary text to the preview buffer.
+3. Add required highlights to the current buffer. If preview buffer is
+   provided, add required highlights to the the preview buffer as well. All
+   highlights must be added to the preview namespace which is provided as an
+   argument to the preview callback (see |nvim_buf_add_highlight()| and
+   |nvim_buf_set_extmark()| for help on how to add highlights to a namespace).
+4. Return an integer which represents the preview type. The preview type
+   tells Nvim what kind of preview is being shown, which allows it to carry
+   out the appropriate tasks to show the preview. There preview type can
+   either be 0, 1 or 2. Here's what the return values mean:
+   0: No preview is shown.
+   1: Preview is shown but preview window isn't opened even if
+      `inccommand=split`.
+   2: Preview is shown and preview window is opened if `inccommand=split`.
+
+   Note that if `inccommand=nosplit`, there's no difference between a return
+   value of 1 and a return value of 2 from the preview callback.
+
+Once preview is over, Nvim will automatically revert any changes made to the
+current buffer and preview buffer during the preview. All highlights set in
+the preview namespace will also be cleared.
+
+Here's an example of a command to trim trailing whitespace from lines that
+supports incremental command preview:
+>
+	-- Trims trailing whitespace from lines
+	-- Shows preview if called as a preview callback
+	-- (which means preview_ns isn't nil)
+	local function TrimTrailingWhitespace(opts, preview_ns, preview_buf)
```

Comment:
- Let's use our Lua conventions in the docs, this means snake_case not PascalCase.
- The dual-purpose of this function doesn't seem to save much code and maybe makes the example less clear?

```suggestion
	-- Trims trailing whitespace in the current buffer.
	-- Also performs 'inccommand' preview if invoked as a preview callback
	-- (preview_ns is non-nil) by editing preview_buf.
	local function trim_space(opts, preview_ns, preview_buf)
```

Reply:
It's actually much simpler to keep them as the same function cause if not we'd need a third function that contains the shared code between the preview function and the command function which I think would only confuse people more.